divi/src/rpcmining.cpp

```diff
@@ -178,6 +180,10 @@ Value setgenerate(const Array& params, bool fHelp)
                 {
                     blockHashes.push_back(chainActive.Tip()->GetBlockHash().GetHex());
                 }
+            } else if (fProofOfStake)
+            {
+                LogPrintf("Failed to generate PoS block, sleeping\n");
+                MilliSleep(1000);
```

Comment:
Yes indeed.  In fact, I've just now done a follow-up (which we can include here or only later where it is needed) commit that makes `setgenerate` throw an RPC error instead.  The test still goes through even with that.  (But it will e.g. later allow us to test for expected errors, for things like mismatches in the block version.)

Reply:
I've added in the follow-up commit now as well to this PR.